packages/react/src/components/createControllerComponent.tsx

```diff
@@ -107,7 +111,11 @@ export const createControllerComponent = <
       // It's also possible for the component to have become unmounted.
       if (this.props.isOpen === true && this.isUnmounted === false) {
         if (this.props.forwardedRef) {
-          (this.props.forwardedRef as any).current = this.overlay;
+          if (typeof this.props.forwardedRef === 'function') {
```

Comment:
Can we extract this to a shared function in the `utils` file? Something like:

```js
const setRef = (ref, value = null) => {
  if (typeof ref === 'function') {
    ref(value);
  } else {
    ref.current = value
  }
};
```

That way we can use this in both `createControllerComponent.tsx` and `createOverlayComponent.tsx`.